sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionClient.java

```diff
@@ -0,0 +1,160 @@
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+
+public class HttpUrlConnectionClient implements HttpClient {
+
+    private HttpRequest request;
+    private HttpURLConnection connection;
+
+    // Default constructor
+    public HttpUrlConnectionClient(){}
+
+    // Constructor initializing with a specific request
+    public HttpUrlConnectionClient(HttpRequest request) {
+        this.setRequest(request);
+    }
+
+    // Setter for HttpRequest
+    public void setRequest(HttpRequest request) {
+        this.request = request;
+    }
+
+    // Asynchronous send method returning a Mono of HttpResponse
+    public Mono<HttpResponse> send() {
+        return sendAsync(this.request, Context.NONE);
+    }
+
+    // Asynchronous send method returning a Mono of HttpResponse
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    // Asynchronous send method returning a Mono of HttpResponse
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    // Synchronous send method with additional context, primarily for interface compliance
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
+    }
+
+    // Asynchronous send method with additional context
+    public Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        return openConnection(httpRequest)
+            .flatMap(connection -> {
+                HttpMethod httpMethod = httpRequest.getHttpMethod();
+
+                if (httpMethod == HttpMethod.PATCH) {
+                    return sendPatchViaSocket(httpRequest);
+                }
+
+                return setConnectionRequest(connection, httpRequest)
+                    .then(writeRequestBody(connection, httpRequest))
+                    .then(readResponse(connection, httpRequest))
+                    .doFinally(signalType -> connection.disconnect()) // Disconnect connection after processing
+                    .onErrorResume(e -> Mono.error(new RuntimeException(e)));
+            });
+    }
+
+    // Send a PATCH request via a SocketClient
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromRunnable(() -> {
+            try {
+                new SocketClient(httpRequest.getUrl().toString()).sendPatchRequest(httpRequest);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }).then(Mono.empty());
+    }
+
+    // Open a connection based on the HttpRequest URL
+    private Mono<HttpURLConnection> openConnection(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> {
+            URL url = new URL(httpRequest.getUrl().toString());
+            return (HttpURLConnection) url.openConnection();
```

Comment:
`HttpRequest.getUrl()` already returns a URL - you do not need to convert it to a String and then parse that String back into a URL. The following should work:

```suggestion
            return (HttpURLConnection) httpRequest.getUrl().openConnection();
```